### PR TITLE
Fix ADLS Gen2 path deletion

### PR DESCRIPTION
**Pull request by stuartleeks**

### CODE REVIEW

File: storage/2018-11-09/datalakestore/paths/delete.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func (client Client) Delete(ctx context.Context, accountName string, fileSystemN
 func (client Client) DeletePreparer(ctx context.Context, accountName string, fileSystemName string, path string) (*http.Request, error) {
 	pathParameters := map[string]interface{}{
 		"fileSystemName": autorest.Encode("path", fileSystemName),
```
Comment (favoretti) [inline storage/2018-11-09/datalakestore/paths/delete.go:46]: Does the fist parameter to Encode have to be "path" as well?

Comment (stuartleeks) [reply storage/2018-11-09/datalakestore/paths/delete.go:46]: Yes. The first parameter indicates the type of encoding (`path` for the `pathParameters`). There are two path parameters, `fileSystemName` and `path`, as seen in the first parameter here:

```go
autorest.WithPathParameters("/{fileSystemName}/{path}", pathParameters)
```

**Comment by favoretti** [comment]
Ah, I see. Thank you for the explanation.

**Comment by stuartleeks** [comment]
No problem. It's funny to look at in this case since path has two meanings!

**Pull request merged**